garrysmod/lua/includes/extensions/util.lua

```diff
@@ -365,3 +365,51 @@ function util.RemovePData( steamid, name )
 	sql.Query( "DELETE FROM playerpdata WHERE infoid = " .. SQLStr( name ) )
 
 end
+
+--[[---------------------------------------------------------
+	Name: StringToBinary( str )
+	Desc: Converts a string to a binary string
+-----------------------------------------------------------]]
+
+function util.StringToBinary( str )
+
+	local bin = {}
+
+	for i = str:len(), 1, -1 do
+		local num = str:byte(i)
+		for j = 7, 0, -1 do -- 8 bits in each char
+			bin[#bin + 1] = math.floor( num / 2 ^ j )
+			num = num % 2 ^ j
+		end
+
+		bin[#bin + 1] = " "
+	end
+
+	return table.concat(bin)
+
+end
+
+--[[---------------------------------------------------------
+	Name: BinaryToString( binary )
+	Desc: Converts a binary string to a readable string
+-----------------------------------------------------------]]
+
+function util.BinaryToString( binary )
+
+	local str = {}
+
+	for byte in binary:gmatch("%S+") do

```

Comment:
This is reliant on each segment of the string being exactly 8 chars which silently errors/gives a garbage result if it is not. Instead, read the string in segments of 8 (use a manually incremented i in a while loop instead of a for loop to be able to skip whitespaces). If the final length without whitespaces is not divisible by 8 or a char other than 0 or 1 show up, error.